util/cpp_compat.h

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides advanced features not available in C++17
+//
+*******************************************************************************/
+
+#include <map>
+#include <unordered_set>
+
+#if __cplusplus >= 202002L
+#define COMPAT_CPP20 [[deprecated]]
+#else
+#define COMPAT_CPP20
+#endif
+
+namespace sycl_cts {
+namespace util {
+namespace details {
+/** @brief Provides implementation for std::erase_if
+ */
+template <class Container, class Pred>
+inline typename Container::size_type erase_if(Container& container, Pred pred) {
+  auto old_size = container.size();
+  // In-place conditional erasure for container
+  for (auto it = container.begin(); it != container.end();) {
+    if (pred(it)) {
+      it = container.erase(it);
+    } else {
+      ++it;
+    }
+  }
+  return old_size - container.size();
+}
+}  // namespace details
+
+/** @brief Provides std::erase_if form std::map available from c++20
```

Comment:
Good idea!
But what do you mean with `form`?

Reply:
`form` was replaced with `from` in this commit: bf40834160c8a1d6e778f94af5f38e06772b05fe